docs/content/FormControl.mdx

```diff
@@ -0,0 +1,429 @@
+
+```jsx live
+<FormControl required>
+  <FormControl.Label>Name</FormControl.Label>
+  <TextInput />
+</FormControl>
+```
+
+### Disabled
+
+```jsx live
+<Box
+  display="flex"
+  flexDirection="column"
+  as="form"
+  sx={{
+    '> * + *': {
+      marginTop: 4
+    }
+  }}
+>
+  <FormControl disabled>
+    <FormControl.Label>Name</FormControl.Label>
+    <TextInput />
+  </FormControl>
+
+  <FormControl variant="choice" disabled>
+    <FormControl.Label>Checkbox option</FormControl.Label>
+    <Checkbox />
+  </FormControl>
+</Box>
+```
+
+### With a visually hidden label
+
+<Note>
+
+Every input must have a corresponding label to be accessible to assistive technology. That's why you'd use `FormControl` instead of using [`TextInput`](/TextInput) directly.
+
+`FormControl` also provides an interface for showing a hint text caption and a validation message, and associating those with the input for assistive technology.
+
+</Note>
+
+```jsx live
+<Box
+  display="flex"
+  flexDirection="column"
+  as="form"
+  sx={{
+    '> * + *': {
+      marginTop: 4
+    }
+  }}
+>
+  <FormControl>
+    <FormControl.Label visuallyHidden>Name</FormControl.Label>
+    <TextInput />
+  </FormControl>
+  <FormControl variant="choice">
+    <FormControl.Label visuallyHidden>Checkbox option</FormControl.Label>
+    <Checkbox />
+  </FormControl>
+</Box>
+```
+
+### With a caption
+
+```jsx live
+<Box
+  display="flex"
+  flexDirection="column"
+  as="form"
+  sx={{
+    '> * + *': {
+      marginTop: 4
+    }
+  }}
+>
+  <FormControl>
+    <FormControl.Label>Name</FormControl.Label>
+    <TextInput />
+    <FormControl.Caption>Hint: your first name</FormControl.Caption>
+  </FormControl>
+  <FormControl variant="choice">
+    <FormControl.Label>Option one</FormControl.Label>
+    <Checkbox />
+    <FormControl.Caption>Hint: the first and only option</FormControl.Caption>
+  </FormControl>
+</Box>
+```
+
+### With validation
+
+<Note>Validation messages are not used for an individual checkbox or radio form control.</Note>
+
+```javascript live noinline
+const ValidationExample = () => {
+  const [value, setValue] = React.useState('mona lisa')
+  const [validationResult, setValidationResult] = React.useState()
+  const doesValueContainSpaces = inputValue => /\s/g.test(inputValue)
+  const handleInputChange = e => {
+    setValue(e.currentTarget.value)
+  }
+
+  React.useEffect(() => {
+    if (doesValueContainSpaces(value)) {
+      setValidationResult('noSpaces')
+    } else if (value) {
+      setValidationResult('validName')
+    }
+  }, [value])
+
+  return (
+    <FormControl
+      validationMap={{
+        noSpaces: 'error',
+        validName: 'success'
+      }}
+      validationResult={validationResult}
+    >
+      <FormControl.Label>GitHub handle</FormControl.Label>
+      <TextInput block value={value} onChange={handleInputChange} />
+      <FormControl.Validation validationKey="noSpaces">GitHub handles cannot contain spaces</FormControl.Validation>
```

Comment:
Would this be a good opportunity to revisit the `validationKey` prop?

Wondering if this pattern - raised at the workshop - is more familiar and intuitive for end users... shall we switch? what do you think @mperrotti?

```suggestion
      {noSpaces && <FormControl.Validation type="success">GitHub handles cannot contain spaces</FormControl.Validation>}
```

Reply:
This was accidentally copied over from the InputField example. I switched over to the pattern you described.